.github/workflows/build_for_riscv.yml

```diff
@@ -69,22 +124,11 @@ jobs:
         shell: /bin/sh
         install: |
           apt-get update -q -y
-          apt-get install -q -y git cmake g++ dpkg
-          apt-get install -q -y software-properties-common
-          apt-get install -q -y llvm-15-dev liblld-15-dev zlib1g-dev
-          apt-get install -q -y wabt # For generating wasm
+          apt-get install -q -y wabt coreutils timeout # For WASM tools and test utilities
         run: |
-          mkdir -p build && cd build
-          cmake -DCMAKE_BUILD_TYPE=Release .. && make -j $(nproc) && make install
-          wasmedge -v
-          wasmedgec -v
-          wasmedge compile -h
-          wasmedge run -h
-          cd ../examples/wasm
-          wat2wasm fibonacci.wat -o fibonacci.wasm
-          wasmedgec fibonacci.wasm fibonacci_aot_c.wasm
-          wasmedge --reactor fibonacci_aot_c.wasm fib 30
-          wasmedge run --reactor fibonacci_aot_c.wasm fib 30
-          wasmedge compile fibonacci.wasm fibonacci_aot_compile.wasm
-          wasmedge --reactor fibonacci_aot_compile.wasm fib 30
-          wasmedge run --reactor fibonacci_aot_compile.wasm fib 30
+          cd /workplace
+          # Set up environment for proper library path propagation
+          export LD_LIBRARY_PATH="/workplace/riscv64-artifacts/lib:${LD_LIBRARY_PATH:-}"
+          # Make test script executable and run tests
+          chmod +x utils/test-riscv-qemu.sh
+          ./utils/test-riscv-qemu.sh /workplace/riscv64-artifacts
```

Comment:
I don't think we need a script to verify the execution; it's too overengineered.